Pod/Classes/PINRemoteImageManager.m

```diff
@@ -1392,6 +1397,29 @@ - (NSString *)cacheKeyForURL:(NSURL *)url processorKey:(NSString *)processorKey
     if (processorKey.length > 0) {
         cacheKey = [cacheKey stringByAppendingString:[NSString stringWithFormat:@"-<%@>", processorKey]];
     }
+
+    //PINDiskCache uses this key as the filename of the file written to disk
+    //Due to the current filesystem used in Darwin, this name must be limited to 255 chars.
+    //In case the generated key exceeds PINRemoteImageManagerCacheKeyMaxLength characters,
+    //we return the hash of it instead.
+    if (cacheKey.length > PINRemoteImageManagerCacheKeyMaxLength) {
+        __block CC_MD5_CTX ctx;
+        CC_MD5_Init(&ctx);
+        NSData *data = [cacheKey dataUsingEncoding:NSUTF8StringEncoding];
+        [data enumerateByteRangesUsingBlock:^(const void * _Nonnull bytes, NSRange byteRange, BOOL * _Nonnull stop) {
+            CC_MD5_Update(&ctx, bytes, (CC_LONG)byteRange.length);
+        }];
+
+        unsigned char digest[CC_MD5_DIGEST_LENGTH];
+        CC_MD5_Final(digest, &ctx);
+
+        NSMutableString *hexString  = [NSMutableString stringWithCapacity:(CC_MD5_DIGEST_LENGTH * 2)];
+        for (int i = 0; i < CC_MD5_DIGEST_LENGTH; i++) {
+            [hexString appendFormat:@"%02lx", (unsigned long)digest[i]];
+        }
+        cacheKey = hexString.copy;
```

Comment:
mind making this cacheKey = [hexString copy]; to match the styling in the rest of the project?